samza-core/src/main/scala/org/apache/samza/config/TaskConfig.scala

```diff
@@ -44,11 +44,13 @@ object TaskConfig {
   val MAX_CONCURRENCY = "task.max.concurrency" // max number of concurrent process for a AsyncStreamTask
   val CALLBACK_TIMEOUT_MS = "task.callback.timeout.ms"  // timeout period for triggering a callback
   val ASYNC_COMMIT = "task.async.commit" // to enable async commit in a AsyncStreamTask
+  val MAX_NO_WORK_WAIT_MS = "task.max.no.work.wait.ms"  // maximum time to wait for a task worker to complete when there are no new messages to handle
```

Comment:
nit: can we change it to task.max.idle.ms?

Reply:
Certainly.  I waffle over naming variables all the time and am happy to
take other people's suggestions.

Somewhat related to a question from @xinyuiscool we could question why I
even made this timeout configurable.  It did not expect anyone to set it to
0 (although I did try that just to see the main loop peg one of the CPUs on
my dev box), but, I suppose I could image a case where you had a box with a
lot of CPUs and were willing to have one spin faster to minimize latency.
The real reason I made in configurable was so it could play with it during
testing without rebuilding my application.